docs/java-helloworld.md

```diff
@@ -0,0 +1,145 @@
+
+Open HelloWorldActivity.java and define the Activity interface:
+
+<!--SNIPSTART java-hello-world-sample-activity-interface-->
+<!--SNIPEND-->
+
+### Workflow
+
+A Workflow is a special function that organizes the sequence of Activity executions. Opposed to an Activity, a Workflow is deterministic in nature. It is where you can specify Activity options, such as what to do if the Activity results in an error or when to time out the Activity execution. Since our Activity is just formatting strings we will only specify the required timeout.
+
+Open HelloWorldWorkflow.java and define the Workflow interface:
+
+<!--SNIPSTART java-hello-world-sample-workflow-interface-->
+<!--SNIPEND-->
+
+Open HelloWorldWorkflowImpl.java and define the Workflow:
+
+<!--SNIPSTART java-hello-world-sample-workflow-->
+<!--SNIPEND-->
+
+### Starter
+
+There are two ways to start a Workflow, either via the CLI or via a starter program. In this tutorial we will start our Workflow by invoking it with a starter program.
+
+Open HelloWorldStarter.java and paste in the following code:
+
+<!--SNIPSTART java-hello-world-sample-workflow-starter-->
+<!--SNIPEND-->
+
+In the code above, we are creating a Temporal client and using it to host the Workflow with the Temporal server. Said another way, this starter program is sending the Workflow to the Temporal server where it will be broken into Tasks. The Tasks will be sent to the Task Queue that we supplied in the Workflow options.
+
+### Worker
+
+A Worker is the process that will actually execute the Workflow and Activity functions as Tasks.
+
+Open HelloWorldWorker.java and paste in the following code:
+
+<!--SNIPSTART java-hello-world-sample-worker-->
+<!--SNIPEND-->
+
+In this function we are creating a Temporal client and using it to create a new Worker. The Worker is registered to handle both HellowWorldWorkflow and HelloWorldActivity Tasks and is configured to listen to the same Task Queue that the Workflow and Activity Tasks are sent to.
+
+## Run "Hello World!" app
+
+At this stage you should have the Temporal server running in a terminal, have the [Temporal Web UI](localhost:8088) open in your browser, and have a project package directory that looks like this:
+
+```
+src/main/java/helloworld/
+  - HelloWorldActivity
+  - HelloWorldActivityImpl
+  - HelloWorldStarter
+  - HelloWorldWorker
+  - HelloWorldWorkflow
+  - HelloWorldWorkflowImpl
+```
+
+Within IntelliJ, right click on HelloWorldStarter and select Run. This will open a terminal inside of IntelliJ which compiles and executes the starter program.
+
+To verify that the Workflow is running, visit the Web UI in your browser and click on the "Workflows" tab. You will see your Workflow listed there with a "running" status. Click on the RunID to see additional details, such as the payload that is being provided to it.
```

Comment:
Is this true? Wont this workflow complete almost instantly and not be running when the user checks Temporal web.

Reply:
Oh I see, I think you intentionally bring up worker second so the workflow is not pulled from a queue and therefore Temporal web will have it in 'running" state.